intelmq/bots/collectors/microsoft/collector_azure.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+"""
+Uses the azure.storage module from https://pypi.python.org/pypi/azure-storage/0.33.0
+Tested with 0.33, probably works with >0.30 too.
+"""
+import datetime
+import gzip
+import io
+
+import pytz
+
+from intelmq.lib.bot import CollectorBot
+
+try:
+    import azure.storage
+except ImportError:
+    azure = None  # noqa
+
+
+class MicrosoftAzureCollectorBot(CollectorBot):
+    def init(self):
+        if azure is None:
+            raise ValueError('Could not import azure.storage. Please install it.')
+
+    def process(self):
+        storage_client = azure.storage.CloudStorageAccount(self.parameters.account_name,
```

Comment:
We should also support proxies settings.

See: def set_proxy(self, host, port, user=None, password=None):
https://azure-storage.readthedocs.io/en/latest/_modules/azure/storage/storageclient.html?highlight=proxy